src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -279,8 +279,26 @@ private void RaiseLockFileTargets()
 
         private void GetPackageAndFileDependencies(LockFileTarget target)
         {
-            var resolvedPackageVersions = target.Libraries
-                .ToDictionary(pkg => pkg.Name, pkg => pkg.Version.ToNormalizedString(), StringComparer.OrdinalIgnoreCase);
+            // In case of there are duplicated packages in libraries with differing versions,
+            // We skip package altogether
+            var resolvedPackageVersions = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var duplicatedPackages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (LockFileTargetLibrary library in target.Libraries)
+            {
+                string libraryName = library.Name;
+                if (duplicatedPackages.Contains(libraryName))
+                {
+                    continue;
+                }
+
+                if (resolvedPackageVersions.ContainsKey(libraryName))
+                {
+                    duplicatedPackages.Add(libraryName);
+                    resolvedPackageVersions.Remove(libraryName);
+                }
```

Comment:
might be better with an else here as you can then avoid having to remove every time